### PR TITLE
Feature/bubble cell custom

### PR DESCRIPTION
**Pull request by kcoleman731**
* This PR creates a new `ATLBaseCollectionViewCell` class which can be used to easily build custom UI on top of Atlas Bubble Cells. 
* Other general code cleanup and refactoring.

### CODE REVIEW

File: Code/Views/ATLBaseCollectionViewCell.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+//  Pods
+//
+//  Created by Kevin Coleman on 12/22/15.
+//
+//
+
+#import <UIKit/UIKit.h>
+#import "ATLMessageBubbleView.h"
+#import "ATLAvatarImageView.h"
+#import "ATLmessagePresenting.h"
```
Comment (kmahal) [inline Code/Views/ATLBaseCollectionViewCell.h]: capital `m`

File: Code/Views/ATLIncomingMessageCollectionViewCell.m [hunk 2]
```diff
@@ -58,15 +58,7 @@ - (id)initWithCoder:(NSCoder *)aDecoder
...
-    [self configureConstraintsForIncomingMessage];
-}
-
-- (void)configureConstraintsForIncomingMessage
-{
-    [self.contentView addConstraint:[NSLayoutConstraint constraintWithItem:self.avatarImageView attribute:NSLayoutAttributeLeft relatedBy:NSLayoutRelationEqual toItem:self.contentView attribute:NSLayoutAttributeLeft multiplier:1.0 constant:ATLAvatarImageLeadPadding]];
-    self.bubbleWithAvatarLeadConstraint = [NSLayoutConstraint constraintWithItem:self.bubbleView attribute:NSLayoutAttributeLeft relatedBy:NSLayoutRelationEqual toItem:self.avatarImageView attribute:NSLayoutAttributeRight multiplier:1.0 constant:ATLAvatarImageTailPadding];
-    [self.contentView addConstraint:self.bubbleWithAvatarLeadConstraint];
-    self.bubbleWithoutAvatarLeadConstraint = [NSLayoutConstraint constraintWithItem:self.bubbleView attribute:NSLayoutAttributeLeft relatedBy:NSLayoutRelationEqual toItem:self.contentView attribute:NSLayoutAttributeLeft multiplier:1.0 constant:ATLMessageCellHorizontalMargin];
+    [self configureCellForType:ATLIncomingCellType];
```
Comment (kmahal) [inline Code/Views/ATLIncomingMessageCollectionViewCell.m]: `super`

File: Code/Views/ATLMessageCollectionViewCell.m [hunk 3]
```diff
@@ -204,34 +174,29 @@ - (void)configureBubbleViewForImageContent
...
-                    // Set self for delegation, if single image message part message
-                    // hasn't been downloaded yet, or is still downloading.
-                    LYRProgress *progress = fullResImagePart.progress;
-                    [progress setDelegate:weakSelf];
-                    weakSelf.progress = progress;
-                    [weakSelf.bubbleView updateProgressIndicatorWithProgress:progress.fractionCompleted visible:YES animated:NO];
-                } else {
-                    [weakSelf.bubbleView updateProgressIndicatorWithProgress:1.0 visible:NO animated:YES];
+        
+        // Fall-back to programatically requesting for a content download of single message part messages (Android compatibillity).
```
Comment (kmahal) [inline Code/Views/ATLMessageCollectionViewCell.m]: talk with Steven, but I think we can get rid of this part now.

File: Code/Views/ATLBaseCollectionViewCell.m [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        [self.contentView removeConstraints:@[self.bubbleViewWidthConstraint]];
+    }
+
+    self.bubbleViewWidthConstraint.constant = bubbleWidth;
+    [self.contentView addConstraint:self.bubbleViewWidthConstraint];
+}
+
+- (void)presentMessage:(LYRMessage *)message
+{
+    self.message = message;
```
Comment (kmahal) [inline Code/Views/ATLBaseCollectionViewCell.m]: where's the code for setting up different cell types based on MIMEtype?

File: Code/Views/ATLBaseCollectionViewCell.h [hunk 5]
```diff
@@ -0,0 +1,67 @@
+//
+//  ATLBaseCollectionViewCell.h
+//  Pods
+//
+//  Created by Kevin Coleman on 12/22/15.
```
Comment (kmahal) [inline Code/Views/ATLBaseCollectionViewCell.h]: Added open source header and remove `Pods` and replace with `Atlas`

File: Code/Views/ATLMessageCollectionViewCell.m (hunk 3 above)
Comment (kmahal) [inline Code/Views/ATLMessageCollectionViewCell.m]: bump

File: Code/Views/ATLIncomingMessageCollectionViewCell.m (hunk 2 above)
Comment (kcoleman731) [inline Code/Views/ATLIncomingMessageCollectionViewCell.m]: No...this subclass does not override the method. Therefore self is fine.

File: Code/Views/ATLBaseCollectionViewCell.m (hunk 4 above)
Comment (kcoleman731) [inline Code/Views/ATLBaseCollectionViewCell.m]: developers take care of it.

**Pull request merged**